airflow/providers/slack/hooks/slack_webhook.py

```diff
@@ -55,6 +55,34 @@ def wrapper(*args, **kwargs) -> Callable:
     return wrapper
 
 
+def _ensure_prefixes(conn_type):
```

Comment:
I can see this function is a duplicate of the one defined in `airflow/providers/slack/hooks/slack.py` (and in many other PR you have open). Would not it be better to put it somewhere common once and then use it across the different providers?

Reply:
you are correct. it's not possible because the only thing providers share is airflow, and we can't use features in airflow until they are present in the version specified as dep in provider. the feature that eliminates the need for this decorator will be released in 2.5.  when min airflow version for these providers is 2.5, then we can remove this decorator. and if you look at the tests, i actually force this removal at this time!  it's tricky managing all the backcompat here but ... that's life!

Reply:
I see. Could you then at least try to have it only once per provider? Here (Slack), you defined this function twice

Reply:
oy, sure